src/server/activeobjectmgr.cpp

```diff
@@ -43,22 +62,31 @@ void ActiveObjectMgr::step(
 		float dtime, const std::function<void(ServerActiveObject *)> &f)
 {
 	g_profiler->avg("ActiveObjectMgr: SAO count [#]", m_active_objects.size());
-	for (auto &ao_it : m_active_objects) {
-		f(ao_it.second);
```

Comment:
Why would this be different between client and server? (on the client we do not collect the IDs first)

Reply:
On the server, there's the `on_step` callback that uses this and can remove and insert objects, which can invalidate iterators. I guess my thought was that such things don't happen on the client, which I'm no longer sure about. I should probably do the same here.